pbrd/pbr_map.c

```diff
@@ -352,7 +352,7 @@ pbr_map_sequence_check_nexthops_valid(struct pbr_map_sequence *pbrms)
 
 static void pbr_map_sequence_check_src_dst_valid(struct pbr_map_sequence *pbrms)
 {
-	if (!pbrms->src && !pbrms->dst)
+	if (!pbrms->src && !pbrms->dst && !pbrms->fwmark)
```

Comment:
Shouldn't we add a new function for fwmark and keep it separate from src/dst? we can then add a new reason for failure, PBR_MAP_INVALID_MARK ?

Reply:
sounds reasonable, however, above code basically checks if anything within pbr-map is set (as before only match on src/dst existed), there is no additional logic to verify if ip address has any reasonable value. So maybe this function should be renamed to something like pbr_map_sequence_check_not_empty  and return reason PBR_MAP_EMPTY on failure, to better reflect its purpose.